totalRP3/modules/map/WorldMapButton.lua

```diff
@@ -0,0 +1,165 @@
+	registerConfigKey(CONFIG_MAP_BUTTON_POSITION, "BOTTOMLEFT");
+
+	local function placeMapButton()
+		local position = getConfigValue(CONFIG_MAP_BUTTON_POSITION)
+
+		WorldMapButton:SetParent(WorldMapFrame.BorderFrame);
+		WorldMapButton:ClearAllPoints();
+
+		local xPadding = 10;
+		local yPadding = 10;
+
+		if position == "TOPRIGHT" then
+			xPadding = -10;
+			yPadding = -45;
+		elseif position == "TOPLEFT" then
+			yPadding = -30;
+		elseif position == "BOTTOMRIGHT" then
+			xPadding = -10;
+			yPadding = 40;
+		end
+
+		WorldMapButton:SetPoint(position, WorldMapFrame.ScrollContainer, position, xPadding, yPadding);
+	end
+
+	insert(TRP3_API.configuration.CONFIG_FRAME_PAGE.elements, {
+		inherit = "TRP3_ConfigH1",
+		title = loc.CO_MAP_BUTTON,
+	});
+
+	insert(TRP3_API.configuration.CONFIG_FRAME_PAGE.elements, {
+		inherit = "TRP3_ConfigDropDown",
+		widgetName = "TRP3_ConfigurationFrame_MapButtonWidget",
+		title = loc.CO_MAP_BUTTON_POS,
+		listContent = {
+			{loc.CO_ANCHOR_BOTTOM_LEFT, "BOTTOMLEFT"},
+			{loc.CO_ANCHOR_TOP_LEFT, "TOPLEFT"},
+			{loc.CO_ANCHOR_BOTTOM_RIGHT, "BOTTOMRIGHT"},
+			{loc.CO_ANCHOR_TOP_RIGHT, "TOPRIGHT"}
+		},
+		listCallback = placeMapButton,
+		listCancel = true,
+		configKey = CONFIG_MAP_BUTTON_POSITION,
+	});
+
+	placeMapButton();
+end)
+--endregion
+
+--region Broadcast Lifecycle
+-- When we get BROADCAST_CHANNEL_CONNECTING we'll ensure the button is
+-- disabled and tell the user things are firing up.
+Events.registerCallback(Events.BROADCAST_CHANNEL_CONNECTING, function()
+	WorldMapButton:SetEnabled(false);
+	WorldMapButton.subtitle = YELLOW(loc.MAP_BUTTON_SUBTITLE_CONNECTING);
+	WorldMapButton.Icon:SetDesaturated(true);
+end);
+
+-- If we get BROADCAST_CHANNEL_OFFLINE we'll ensure the button remains
+-- disabled and dump the localised error into the tooltip, to be useful.
+Events.registerCallback(Events.BROADCAST_CHANNEL_OFFLINE, function(reason)
+	WorldMapButton:SetEnabled(false);
+	WorldMapButton.subtitle = YELLOW(loc.MAP_BUTTON_SUBTITLE_OFFLINE):format(reason);
+	WorldMapButton.Icon:SetDesaturated(true);
+end);
+
+-- When we get BROADCAST_CHANNEL_READY it's time to enable the button use the
+-- standard tooltip description.
+Events.registerCallback(Events.BROADCAST_CHANNEL_READY, function()
+	WorldMapButton:SetEnabled(true);
+	WorldMapButton.subtitle = YELLOW(loc.MAP_BUTTON_SUBTITLE);
+	WorldMapButton.Icon:SetDesaturated(false);
+end);
+
+--endregion
+
+--region UI setup
+setupIconButton(WorldMapButton, "icon_treasuremap");
+---@type Tooltip
+Ellyb.Tooltips.getTooltip(WorldMapButton)
+	 :SetTitle(loc.MAP_BUTTON_TITLE)
+	 :OnShow(function(tooltip)
+	tooltip:AddTempLine(WorldMapButton.subtitle)
+end)
+--endregion
+
+WorldMapButton:SetScript("OnClick", function(self)
+	local structure = {};
+	for scanID, scan in pairs(TRP3_API.MapScannersManager.getAllScans()) do
+		if scan:CanScan() then
+			insert(structure, { icon(scan.scanIcon, 20) .. " " .. (scan.scanOptionText or scanID), scanID});
```

Comment:
Space after the `{` but none before the following `}`. The next occurrence has no surrounding spaces at all. This triggers me immensely.